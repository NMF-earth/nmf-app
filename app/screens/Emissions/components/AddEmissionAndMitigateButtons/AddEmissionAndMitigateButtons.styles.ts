import { StyleSheet } from "react-native";

import { PADDING_HORIZONTAL } from "../../../../constants/Layout";

const button = {
  flex: 1,
  marginBottom: 20,
};
export default StyleSheet.create({
  buttonView: {
    flexDirection: "row",
    position: "absolute",
    bottom: 0,
    right: 0,
    left: 0,
  },
  buttonLeft: {
    ...button,
    marginLeft: PADDING_HORIZONTAL,
    marginRight: PADDING_HORIZONTAL / 2,
  },
  buttonRight: {
    ...button,
    marginRight: PADDING_HORIZONTAL,
    marginLeft: PADDING_HORIZONTAL / 2,
  },
});
