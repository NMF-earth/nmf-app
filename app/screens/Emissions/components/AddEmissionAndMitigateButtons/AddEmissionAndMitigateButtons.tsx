import React from "react";
import { View } from "react-native";
import { t } from "../../../../utils";
import styles from "./AddEmissionAndMitigateButtons.styles";
import { Text, Button } from "../../../../components";

interface Props {
  navigation: {
    push: (screen: string) => void;
    navigate: (screen: string) => void;
  };
}

const AddEmissionAndMitigateButtons = ({ navigation }: Props) => (
  <View style={styles.buttonView}>
    <Button.Secondary
      style={styles.buttonLeft}
      onPress={() => navigation.navigate("ComingSoon")}
      textType={"Secondary"}
    >
      <Text.Secondary center bold green>
        {t("EMISSIONS_SCREEN_MITIGATE")}
      </Text.Secondary>
    </Button.Secondary>
    <Button.Primary
      style={styles.buttonRight}
      onPress={() => {
        navigation.push("AddEmission");
      }}
      textType={"Secondary"}
    >
      <Text.Secondary center white bold>
        {t("EMISSIONS_SCREEN_ADD")}
      </Text.Secondary>
    </Button.Primary>
  </View>
);

export default AddEmissionAndMitigateButtons;
