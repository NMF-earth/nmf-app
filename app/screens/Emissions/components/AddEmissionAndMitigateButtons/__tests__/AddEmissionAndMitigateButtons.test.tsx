import React from "react";
import renderer from "react-test-renderer";
import AddEmissionAndMitigateButtons from "../AddEmissionAndMitigateButtons";

const props = {
  navigation: {
    push: jest.fn(),
    navigate: jest.fn()
  }
};

jest.unmock("../AddEmissionAndMitigateButtons");

it("AddEmissionAndMitigateButtons renders correctly", () => {
  const tree = renderer
    .create(<AddEmissionAndMitigateButtons {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("AddEmissionAndMitigateButtons should navigate if any button is pressed", () => {
  const root = renderer.create(<AddEmissionAndMitigateButtons {...props} />)
    .root;
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  const buttonPrimary = root.findByType("Button.Primary");
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  const buttonSecondary = root.findByType("Button.Secondary");

  buttonPrimary.props.onPress();
  buttonSecondary.props.onPress();
  expect(props.navigation.navigate).toHaveBeenCalled();
  expect(props.navigation.push).toHaveBeenCalled();
});
