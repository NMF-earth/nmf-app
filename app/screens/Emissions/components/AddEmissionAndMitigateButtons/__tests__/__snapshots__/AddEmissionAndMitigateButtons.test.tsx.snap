// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEmissionAndMitigateButtons renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "flexDirection": "row",
      "left": 0,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <Button.Secondary
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
        "marginBottom": 20,
        "marginLeft": 14,
        "marginRight": 7,
      }
    }
    textType="Secondary"
  >
    <Text.Secondary
      bold={true}
      center={true}
      green={true}
    >
      Mitigate
    </Text.Secondary>
  </Button.Secondary>
  <Button.Primary
    onPress={[Function]}
    style={
      Object {
        "flex": 1,
        "marginBottom": 20,
        "marginLeft": 7,
        "marginRight": 14,
      }
    }
    textType="Secondary"
  >
    <Text.Secondary
      bold={true}
      center={true}
      white={true}
    >
      Add
    </Text.Secondary>
  </Button.Primary>
</View>
`;
